File: 2785. Sort Vowels in a String.cpp
class Solution {
public:
    string sortVowels(string s) 
    {
        string v;
        int n = s.size();
        for(int i=0;i<n;i++)
        {
            char ch = s[i];
            if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U') v+=ch;
        }

        if(v.empty()) return s;
        
        int k = 0;
        sort(v.begin(),v.end());
        for(int i=0;i<n;i++)
        {
            char &ch = s[i];
            if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U') 
            {
                ch = v[k];
                k++;
            }
        }
        return s;
    }
};

File: String/2785. Sort Vowels in a String.py
class Solution:
    def is_vowel(self, char):
        return char.lower() in 'aeiou'
    def sortVowels(self, s: str) -> str:
        ans = [ch  for ch in s]
        vowel = [ch for ch in s if self.is_vowel(ch)]
        vowel.sort()
        
        n, cur = len(ans), 0
        for i in range(n):
            if self.is_vowel(s[i]):
                ans[i] = vowel[cur]
                cur += 1
        return ''.join(ans)